miasm2/jitter/jitload.py

```diff
@@ -257,30 +258,47 @@ def exception_memory_breakpoint(jitter):
         self.add_exception_handler(EXCEPT_BREAKPOINT_INTERN,
                                    exception_memory_breakpoint)
 
-    def add_breakpoint(self, addr, callback):
-        """Add a callback associated with addr.
+    def set_breakpoint(self, addr, callback):
+        """Add a berakpont at addr to @callback.
+        WARNING: an address can have only one callback
         @addr: breakpoint address
         @callback: function with definition (jitter instance)
         """
-        self.breakpoints_handler.add_callback(addr, callback)
+
+        if addr in self.breakpoints_handlers:
+            raise ValueError("A breakpoint is already set at %x" % addr)
```

Comment:
It could be great for the user to have a pattern to actually set a breakpoint even if it already exists, or at least easily detect it.
Maybe:
* A custom error `BreakpointAlreadyExist` instead of the generic `ValueError`, which may be raised from other code part
* A `force` option, by default to False, which force the breakpoint update. It will avoid the user to directly manipulate `.breakpoints_handlers` internal structure by forcing its changes through a dedicated API (BTW, this internal structure could become protected)